### ISSUE TITLE
[2.1.1] [OS X] [Astrill VPN client] Connection errors "Connection timed out"

### ISSUE THREAD

**Issue by pjrobertson** [owncloud/client#4530]
### Expected behaviour

The OS X client should continue to sync with the server despite network changes
### Actual behaviour

When a user enables/disabled their VPN and the system network changes, the owncloud client reports an error connecting to the server until it is restarted. Changing the proxy settings in the 'Network' settings panel to 'No proxy' does not fix the issue.
An Astrill VPN client is being used to connect to a VPN and change the network settings.
### Steps to reproduce
1. Start client and wait for successful sync
2. Turn on VPN, then turn it off (I am not sure on the exact requirements)
3. Open owncloud client and note that it now cannot sync
4. Restart the client for everything to start working again
### Server configuration

Operating system:
owncloud 8.2.2 running on CentOS 7, PHP7, MySQL
### Client configuration

Client version:
Version 2.1.1

Operating system:
Mac OS X 10.10

OS language:
Chinese
### Logs

If these are required I can get them (it is a colleague's computer that is having issues)

**Comment by guruz** [comment]
What is the _exact_ error you are getting? How does this look in the UI? Are you sure the client does not reconnect after some minutes?

**Comment by pjrobertson** [comment]
I just got this in the UI. If I pause/restart the connection it still doesn't work. Changing my internet connection also doesn't solve anything. There's nothing in console.app

![screen shot 2016-03-03 at 23 03 11](https://cloud.githubusercontent.com/assets/150431/13513540/8c4e15c4-e1dc-11e5-9488-55b6f051081e.png)

**Comment by guruz** [comment]
Have you tried asking the Astrill support?

**Comment by pjrobertson** [comment]
I can confirm this isn't specific to Astrill, but for any app that alters the proxy setting on the computer. Other VPN apps have the same problem. Even turning the Network settings in Owncloud to 'no proxy' still does not fix the problem

**Comment by guruz** [comment]
So the proxy _DOES_ change? How does it look like in the OS X settings? Do you think the problem is that the client does not detect and use the proxy?

**Comment by pjrobertson** [comment]
Yes, so the VPN changes the OS X proxy settings as it is enabled/disabled. That's why I changed the Owncloud settings to read 'no proxy'. In this case it shouldn't matter to the client whether or not the system is using a proxy

**Comment by guruz** [comment]
So do you think the problem is actually the changing of the network used? No matter which network or VPN software?

Would you be open to testing a recent nightly build of the OS X client? We have some changes regarding network connectivity in there.. 

http://download.owncloud.com/desktop/daily/ownCloud-2.2.0.3281-nightly20160419.pkg

**Comment by pjrobertson** [comment]
Yes, I think the problem occurs when the VPN software (whichever one it is) changes the network connections. the ownCloud client probably detects some change and then breaks.

Sure, I'll test out the nightly build. I'm testing it now
